tools/system_libs.py

```diff
@@ -210,6 +220,8 @@ def join(flags):
     out += f'build {libname}: archive {objects}\n'
 
   utils.write_file(filename, out)
+  parent_ninja = os.path.join(os.path.dirname(build_dir), 'build.ninja')
```

Comment:
Seems like we are assuming here that this file will match the one in build_deferred below which is calculated via  `cache.get_path('build')`?

Can you use `os.path.join(cache.get_path('build'), 'build.ninja'` here too?    Or create a function to get this? e.g. `top_level_ninja_file()`.

Reply:
Nevermind, the top-level ninja is separable from the individual library ninjas (those functions just pass the build dir around between them). So now the top-level ninja dir is centralized as you suggested.